sessiontxn/staleread/processor.go

```diff
@@ -179,7 +182,7 @@ func (p *staleReadProcessor) OnSelectTable(tn *ast.TableName) error {
 		}
 		return nil
 	}
-	return p.evaluateFromStmtTSOrSysVariable(stmtAsOfTS)
+	return p.evaluateFromStmtTSOrSysVariable(stmtAsOfTS, evaluateTS)
```

Comment:
What's the difference between `OnExecutePreparedStmt` and `OnSelectTable`, why do we need to calculate the `TS` in two places? Is it possible to merge them into one?

Reply:
Here we set the ts evaluator which needs to be calculated to staleness processor when preparing, and it'll be returned and set to plan cache.

When invoking `OnExecutePreparedStmt`, it only uses the ts evaluator from plan cache, so it's not necessary to set a evaluator that requires calculation to get a timestamp.